### PR TITLE
HBASE-24810: Download Boost with other dependencies

### PR DESCRIPTION
**Pull request by bharathv**

### CODE REVIEW

**Comment by Apache-HBase** [comment]
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://builds.apache.org/job/hbase-native-client-github-precommit/job/PR-12/1/console in case of problems.

**Comment by bharathv** [comment]
@joshelser @phrocker FYI

With this change, all the tests pass for me locally since the boost dependency issue is fixed. Lets see what the precommit has to say.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 315 | Docker mode activated. |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| -1 | CMakeLists.txt | 1060 | root in master failed. |
| -1 | compile | 6 | root in master failed. |
||| _ Patch Compile Tests _ |
| +1 | CMakeLists.txt | 278 | the patch passed |
| -1 | compile | 361 | root in the patch failed. |
||| _ Other Tests _ |
| -1 | unit | 31 | root in the patch failed. |
| | | 2069 | |


| Reason | Tests |
|-------:|:------|
| Failed CTEST tests | hbase-configuration-test |
|   | server-name-test |
|   | client-deserializer-test |
|   | location-cache-test |
|   | concurrent-map-test |
|   | rpc-test |
|   | scanner-test |
|   | cell-test |
|   | result-test |
|   | connection-pool-test |
|   | scan-result-cache-test |
|   | location-cache-retry-test |
|   | append-test |
|   | zk-util-test |
|   | get-test |
|   | user-util-test |
|   | region-info-deserializer-test |
|   | time-range-test |
|   | bytes-util-test |
|   | client-serializer-test |
|   | async-rpc-retrying-test |
|   | scan-test |
|   | put-test |
|   | configuration-test |
|   | request-converter-test |
|   | client-test |
|   | table-name-test |
|   | filter-test |
|   | increment-test |
|   | delete-test |
|   | async-batch-rpc-retrying-test |
|   | zk-deserializer-test |
|   | exception-test |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://builds.apache.org/job/hbase-native-client-github-precommit/job/PR-12/1/artifact/hbase-native-client-precommit/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-native-client/pull/12 |
| JIRA Issue | HBASE-24810 |
| Optional Tests | compile unit |
| uname | Linux securecluster 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | cmake |
| Personality | bin/hbase-native-client-personality.sh |
| git revision | master / 7281fc6 |
| CMakeLists.txt | https://builds.apache.org/job/hbase-native-client-github-precommit/job/PR-12/1/artifact/hbase-native-client-precommit/output/branch-CMakeLists.txt-root.txt |
| compile | https://builds.apache.org/job/hbase-native-client-github-precommit/job/PR-12/1/artifact/hbase-native-client-precommit/output/branch-compile-root.txt |
| compile | https://builds.apache.org/job/hbase-native-client-github-precommit/job/PR-12/1/artifact/hbase-native-client-precommit/output/patch-compile-root.txt |
| CTEST | https://builds.apache.org/job/hbase-native-client-github-precommit/job/PR-12/1/artifact/hbase-native-client-precommit/output/patch-root-ctest.txt |
| unit | https://builds.apache.org/job/hbase-native-client-github-precommit/job/PR-12/1/artifact/hbase-native-client-precommit/output/patch-unit-root.txt |
|  Test Results | https://builds.apache.org/job/hbase-native-client-github-precommit/job/PR-12/1/testReport/ |
| Max. process+thread count | 111 (vs. ulimit of 1000) |
| modules | C: . U: . |
| Console output | https://builds.apache.org/job/hbase-native-client-github-precommit/job/PR-12/1/console |
| versions | git=2.7.4 cmake=cmake version 3.5.1 make=GNU Make 4.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by bharathv** [comment]
Looks like a g++ crash due to memory issue, especially after my last change that does a multi-threaded make. Let me try bumping up the docker mem limit.

`g++: internal compiler error: Killed (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-5/README.Bugs> for instructions.
Makefile:1965: recipe for target 'experimental/JSONSchema.lo' failed`

**Comment by Apache-HBase** [comment]
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/1/console in case of problems.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 932 | Docker mode activated. |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| -1 | CMakeLists.txt | 1056 | root in master failed. |
| -1 | compile | 6 | root in master failed. |
||| _ Patch Compile Tests _ |
| +1 | CMakeLists.txt | 291 | the patch passed |
| -1 | compile | 340 | root in the patch failed. |
||| _ Other Tests _ |
| -1 | unit | 20 | root in the patch failed. |
| | | 2661 | |


| Reason | Tests |
|-------:|:------|
| Failed CTEST tests | location-cache-test |
|   | filter-test |
|   | async-rpc-retrying-test |
|   | location-cache-retry-test |
|   | hbase-configuration-test |
|   | table-name-test |
|   | delete-test |
|   | client-test |
|   | client-serializer-test |
|   | scan-test |
|   | request-converter-test |
|   | concurrent-map-test |
|   | zk-util-test |
|   | result-test |
|   | exception-test |
|   | user-util-test |
|   | server-name-test |
|   | zk-deserializer-test |
|   | get-test |
|   | put-test |
|   | configuration-test |
|   | increment-test |
|   | cell-test |
|   | connection-pool-test |
|   | append-test |
|   | bytes-util-test |
|   | scanner-test |
|   | time-range-test |
|   | scan-result-cache-test |
|   | client-deserializer-test |
|   | rpc-test |
|   | region-info-deserializer-test |
|   | async-batch-rpc-retrying-test |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/1/artifact/hbase-native-client-precommit/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-native-client/pull/12 |
| JIRA Issue | HBASE-24810 |
| Optional Tests | compile unit |
| uname | Linux securecluster 4.15.0-65-generic #74-Ubuntu SMP Tue Sep 17 17:06:04 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | cmake |
| Personality | bin/hbase-native-client-personality.sh |
| git revision | master / 7281fc6 |
| CMakeLists.txt | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/1/artifact/hbase-native-client-precommit/output/branch-CMakeLists.txt-root.txt |
| compile | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/1/artifact/hbase-native-client-precommit/output/branch-compile-root.txt |
| compile | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/1/artifact/hbase-native-client-precommit/output/patch-compile-root.txt |
| CTEST | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/1/artifact/hbase-native-client-precommit/output/patch-root-ctest.txt |
| unit | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/1/artifact/hbase-native-client-precommit/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/1/testReport/ |
| Max. process+thread count | 111 (vs. ulimit of 1000) |
| modules | C: . U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/1/console |
| versions | git=2.7.4 cmake=cmake version 3.5.1 make=GNU Make 4.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/2/console in case of problems.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 317 | Docker mode activated. |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| -1 | CMakeLists.txt | 1057 | root in master failed. |
| -1 | compile | 6 | root in master failed. |
||| _ Patch Compile Tests _ |
| +1 | CMakeLists.txt | 281 | the patch passed |
| -1 | compile | 581 | root in the patch failed. |
||| _ Other Tests _ |
| -1 | unit | 21 | root in the patch failed. |
| | | 2281 | |


| Reason | Tests |
|-------:|:------|
| Failed CTEST tests | exception-test |
|   | scan-test |
|   | request-converter-test |
|   | async-rpc-retrying-test |
|   | scanner-test |
|   | location-cache-test |
|   | rpc-test |
|   | client-deserializer-test |
|   | increment-test |
|   | delete-test |
|   | location-cache-retry-test |
|   | zk-util-test |
|   | bytes-util-test |
|   | server-name-test |
|   | concurrent-map-test |
|   | get-test |
|   | user-util-test |
|   | cell-test |
|   | table-name-test |
|   | configuration-test |
|   | region-info-deserializer-test |
|   | hbase-configuration-test |
|   | zk-deserializer-test |
|   | client-test |
|   | async-batch-rpc-retrying-test |
|   | scan-result-cache-test |
|   | put-test |
|   | connection-pool-test |
|   | append-test |
|   | time-range-test |
|   | client-serializer-test |
|   | result-test |
|   | filter-test |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/2/artifact/hbase-native-client-precommit/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-native-client/pull/12 |
| JIRA Issue | HBASE-24810 |
| Optional Tests | compile unit |
| uname | Linux securecluster 4.15.0-60-generic #67-Ubuntu SMP Thu Aug 22 16:55:30 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | cmake |
| Personality | bin/hbase-native-client-personality.sh |
| git revision | master / 7281fc6 |
| CMakeLists.txt | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/2/artifact/hbase-native-client-precommit/output/branch-CMakeLists.txt-root.txt |
| compile | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/2/artifact/hbase-native-client-precommit/output/branch-compile-root.txt |
| compile | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/2/artifact/hbase-native-client-precommit/output/patch-compile-root.txt |
| CTEST | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/2/artifact/hbase-native-client-precommit/output/patch-root-ctest.txt |
| unit | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/2/artifact/hbase-native-client-precommit/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/2/testReport/ |
| Max. process+thread count | 75 (vs. ulimit of 1000) |
| modules | C: . U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/2/console |
| versions | git=2.7.4 cmake=cmake version 3.5.1 make=GNU Make 4.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: cmake/DownloadBoost.cmake [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  # Only compile and install the needed libs.
+  set(LIBS_TO_COMPILE "")
+  foreach(lib ${BOOST_LIBS})
+    string(APPEND LIBS_TO_COMPILE --with-${lib} " ")
+  endforeach()
+
+  separate_arguments(BUILD_CMD UNIX_COMMAND "./b2 ${LIBS_TO_COMPILE} install")
+
+  ExternalProject_Add(boost
+     URL "https://dl.bintray.com/boostorg/release/1.65.1/source/boost_1_65_1.tar.gz"
```
Comment (phrocker) [inline cmake/DownloadBoost.cmake:41]: Does this build a static lib?

File: cmake/boost/local/FindBoost.cmake [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+# under the License.
+
+# Stubs to allow us to find Boost libs
+
+set(Boost_INCLUDE_DIRS "${BOOST_ROOT}/include" CACHE STRING "" FORCE)
+set(Boost_INCLUDE_DIR "${BOOST_ROOT}/include" CACHE STRING "" FORCE)
+
+set(Boost_LIBRARIES "" CACHE STRING "" FORCE)
+foreach(COMPONENT ${Boost_FIND_COMPONENTS})
+  list(APPEND Boost_LIBRARIES "${BOOST_ROOT}/lib/${BYPRODUCT_PREFIX}boost_${COMPONENT}${BYPRODUCT_SHARED_SUFFIX}")
```
Comment (phrocker) [inline cmake/boost/local/FindBoost.cmake]: Given BYPRODUCT_SHARED_SUFFIX, will the install ensure these shared objects are available for the distributed binary?

File: cmake/DownloadBoost.cmake [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+  # Only compile and install the needed libs.
+  set(LIBS_TO_COMPILE "")
+  foreach(lib ${BOOST_LIBS})
+    string(APPEND LIBS_TO_COMPILE --with-${lib} " ")
+  endforeach()
+
+  separate_arguments(BUILD_CMD UNIX_COMMAND "./b2 ${LIBS_TO_COMPILE} install")
+
+  ExternalProject_Add(boost
```
Comment (phrocker) [inline cmake/DownloadBoost.cmake:40]: Does this significantly add to the build time? I haven't pulled it down to try but will.

File: cmake/DownloadBoost.cmake [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+     URL "https://dl.bintray.com/boostorg/release/1.65.1/source/boost_1_65_1.tar.gz"
+     PREFIX "${BUILD_DIR}/dependencies"
+     DOWNLOAD_DIR ${BOOST_DOWNLOAD_DIR}
+     BUILD_IN_SOURCE true
+     SOURCE_DIR ${BOOST_SOURCE_DIR}
+     INSTALL_DIR ${BOOST_INSTALL_DIR}
+     CONFIGURE_COMMAND ./bootstrap.sh --prefix=${BOOST_INSTALL_DIR}
```
Comment (phrocker) [inline cmake/DownloadBoost.cmake:47]: You could probably statically link against boost. Not sure that is of interest I guess it's a balance of size

File: cmake/DownloadBoost.cmake (hunk 3 above)
Comment (bharathv) [reply cmake/DownloadBoost.cmake:40]: Not really actually, a few more mins, I think.

File: cmake/DownloadBoost.cmake (hunk 1 above)
Comment (bharathv) [reply cmake/DownloadBoost.cmake:41]: It builds both. Actually I was only trying to glue the whole thing together and haven't thought much about boost compiling options. Let me fix this. I think we can just build a static/mt version/release build.

File: cmake/boost/local/FindBoost.cmake (hunk 2 above)
Comment (bharathv) [reply cmake/boost/local/FindBoost.cmake]: This was unintentional, the build was building both shared and static libs and hence this worked. Let me move to static libs.

**Comment by Apache-HBase** [comment]
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/4/console in case of problems.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 324 | Docker mode activated. |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| -1 | CMakeLists.txt | 1061 | root in master failed. |
| -1 | compile | 6 | root in master failed. |
||| _ Patch Compile Tests _ |
| +1 | CMakeLists.txt | 290 | the patch passed |
| -1 | compile | 528 | root in the patch failed. |
||| _ Other Tests _ |
| -1 | unit | 20 | root in the patch failed. |
| | | 2247 | |


| Reason | Tests |
|-------:|:------|
| Failed CTEST tests | scan-test |
|   | request-converter-test |
|   | filter-test |
|   | client-test |
|   | hbase-configuration-test |
|   | time-range-test |
|   | exception-test |
|   | user-util-test |
|   | delete-test |
|   | bytes-util-test |
|   | client-deserializer-test |
|   | region-info-deserializer-test |
|   | rpc-test |
|   | zk-deserializer-test |
|   | scan-result-cache-test |
|   | configuration-test |
|   | append-test |
|   | scanner-test |
|   | cell-test |
|   | location-cache-retry-test |
|   | concurrent-map-test |
|   | table-name-test |
|   | get-test |
|   | zk-util-test |
|   | put-test |
|   | client-serializer-test |
|   | async-batch-rpc-retrying-test |
|   | location-cache-test |
|   | increment-test |
|   | connection-pool-test |
|   | server-name-test |
|   | result-test |
|   | async-rpc-retrying-test |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/4/artifact/hbase-native-client-precommit/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-native-client/pull/12 |
| JIRA Issue | HBASE-24810 |
| Optional Tests | compile unit |
| uname | Linux securecluster 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | cmake |
| Personality | bin/hbase-native-client-personality.sh |
| git revision | master / 3b86174 |
| CMakeLists.txt | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/4/artifact/hbase-native-client-precommit/output/branch-CMakeLists.txt-root.txt |
| compile | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/4/artifact/hbase-native-client-precommit/output/branch-compile-root.txt |
| compile | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/4/artifact/hbase-native-client-precommit/output/patch-compile-root.txt |
| CTEST | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/4/artifact/hbase-native-client-precommit/output/patch-root-ctest.txt |
| unit | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/4/artifact/hbase-native-client-precommit/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/4/testReport/ |
| Max. process+thread count | 75 (vs. ulimit of 1000) |
| modules | C: . U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/4/console |
| versions | git=2.7.4 cmake=cmake version 3.5.1 make=GNU Make 4.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/5/console in case of problems.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 27 | Docker mode activated. |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| -1 | CMakeLists.txt | 1061 | root in master failed. |
| -1 | compile | 6 | root in master failed. |
||| _ Patch Compile Tests _ |
| +1 | CMakeLists.txt | 286 | the patch passed |
| +1 | compile | 711 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 251 | root in the patch failed. |
| | | 2361 | |


| Reason | Tests |
|-------:|:------|
| Failed CTEST tests | client-test |
|   | scanner-test |
|   | async-batch-rpc-retrying-test |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/5/artifact/hbase-native-client-precommit/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-native-client/pull/12 |
| JIRA Issue | HBASE-24810 |
| Optional Tests | compile unit |
| uname | Linux securecluster 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | cmake |
| Personality | bin/hbase-native-client-personality.sh |
| git revision | master / 3b86174 |
| CMakeLists.txt | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/5/artifact/hbase-native-client-precommit/output/branch-CMakeLists.txt-root.txt |
| compile | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/5/artifact/hbase-native-client-precommit/output/branch-compile-root.txt |
| CTEST | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/5/artifact/hbase-native-client-precommit/output/patch-root-ctest.txt |
| unit | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/5/artifact/hbase-native-client-precommit/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/5/testReport/ |
| Max. process+thread count | 914 (vs. ulimit of 1000) |
| modules | C: . U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/5/console |
| versions | git=2.7.4 cmake=cmake version 3.5.1 make=GNU Make 4.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Review by bharathv — COMMENTED** [review]
Patch is ready for review. 

After trying out a few make concurrency values, we finally have a working test run with 3 of 33 tests failing (they pass for me locally though) yay!!. I enabled failed test logging so that we have the test logs in the artifacts.

**Comment by Apache-HBase** [comment]
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/6/console in case of problems.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 28 | Docker mode activated. |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| -1 | CMakeLists.txt | 1063 | root in master failed. |
| -1 | compile | 6 | root in master failed. |
||| _ Patch Compile Tests _ |
| +1 | CMakeLists.txt | 284 | the patch passed |
| +1 | compile | 712 | the patch passed |
||| _ Other Tests _ |
| -1 | unit | 249 | root in the patch failed. |
| | | 2359 | |


| Reason | Tests |
|-------:|:------|
| Failed CTEST tests | client-test |
|   | scanner-test |
|   | async-batch-rpc-retrying-test |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/6/artifact/hbase-native-client-precommit/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-native-client/pull/12 |
| JIRA Issue | HBASE-24810 |
| Optional Tests | compile unit |
| uname | Linux securecluster 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | cmake |
| Personality | bin/hbase-native-client-personality.sh |
| git revision | master / 3b86174 |
| CMakeLists.txt | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/6/artifact/hbase-native-client-precommit/output/branch-CMakeLists.txt-root.txt |
| compile | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/6/artifact/hbase-native-client-precommit/output/branch-compile-root.txt |
| CTEST | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/6/artifact/hbase-native-client-precommit/output/patch-root-ctest.txt |
| unit | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/6/artifact/hbase-native-client-precommit/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/6/testReport/ |
| Max. process+thread count | 969 (vs. ulimit of 1000) |
| modules | C: . U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/6/console |
| versions | git=2.7.4 cmake=cmake version 3.5.1 make=GNU Make 4.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by joshelser** [comment]
Wow! Lots of progress since I've last made some time. I was able to validate that building works from within the provided docker container (e.g. `cmake -DDOWNLOAD_DEPENDENCIES=ON .`). I did see some test failures after,

```
The following tests FAILED:
	 15 - client-test (OTHER_FAULT)
	 16 - async-batch-rpc-retrying-test (OTHER_FAULT)
	 18 - scanner-test (OTHER_FAULT)
	 24 - location-cache-retry-test (OTHER_FAULT)
	 26 - filter-test (OTHER_FAULT)
	 30 - location-cache-test (OTHER_FAULT)
	 32 - async-rpc-retrying-test (OTHER_FAULT)
```

I'm guessing these are trying to talk to an HBase which I don't have running locally, so I'm not too concerned ;)

**Review by joshelser — APPROVED** [review]
Looks great to this c++ noob.

**Comment by bharathv** [comment]
Thanks for taking a look @joshelser.

> I'm guessing these are trying to talk to an HBase which I don't have running locally, so I'm not too concerned.

No, we do have an in-process mini cluster that runs via JNI (its a simple wrapper over the Java implementation). In my local setup all the tests pass for me. If it is not too much effort, can you please share the stdout of one of the failing tests? Thanks.

**Comment by bharathv** [comment]
@phrocker you have any more comments? I think I addressed your concerns but it'd be great if you can double check. Thanks.

**Comment by Apache-HBase** [comment]
(!) A patch to the testing environment has been detected. 
Re-executing against the patched versions to perform further tests. 
The console is at https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/7/console in case of problems.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 638 | Docker mode activated. |
||| _ Prechecks _ |
||| _ master Compile Tests _ |
| -1 | CMakeLists.txt | 15 | root in master failed. |
| -1 | compile | 15 | root in master failed. |
||| _ Patch Compile Tests _ |
| -1 | CMakeLists.txt | 16 | root in the patch failed. |
| -1 | compile | 15 | root in the patch failed. |
||| _ Other Tests _ |
| -1 | unit | 15 | root in the patch failed. |
| | | 735 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.40 ServerAPI=1.40 base: https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/7/artifact/hbase-native-client-precommit/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase-native-client/pull/12 |
| JIRA Issue | HBASE-24810 |
| Optional Tests | compile unit |
| uname | Linux securecluster 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | cmake |
| Personality | bin/hbase-native-client-personality.sh |
| git revision | master / 3b86174 |
| CMakeLists.txt | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/7/artifact/hbase-native-client-precommit/output/branch-CMakeLists.txt-root.txt |
| compile | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/7/artifact/hbase-native-client-precommit/output/branch-compile-root.txt |
| CMakeLists.txt | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/7/artifact/hbase-native-client-precommit/output/patch-CMakeLists.txt-root.txt |
| compile | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/7/artifact/hbase-native-client-precommit/output/patch-compile-root.txt |
| unit | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/7/artifact/hbase-native-client-precommit/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/7/testReport/ |
| Max. process+thread count | 8 (vs. ulimit of 1000) |
| modules | C: . U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/hbase-native-client-github-precommit/job/PR-12/7/console |
| versions | git=2.7.4 cmake=cmake version 3.5.1 make=GNU Make 4.1 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by joshelser** [comment]
Talked to Marc briefly today. He said he had a busy week, but he is meaning to come back and take a look. You're probably OK to move ahead if you have time to work on this, Bharath :)

**Comment by bharathv** [comment]
Thanks @joshelser. Let me merge this then. @phrocker I think we can revisit these pieces again once you rebase #6 on top of this patch. Eventually we need to make another pass over all the dependencies and make sure everything is being built with the right flags and linked correctly. For now I think this is in a decent shape and passing most tests, so getting this in.

**Pull request merged**